services/core-web/src/components/noticeOfWork/applications/referals/ReferralConsultationPackage.js

```diff
@@ -0,0 +1,249 @@
+        this.props.setNoticeOfWorkApplicationDocumentDownloadState({
+          downloading: true,
+          currentFile,
+          totalFiles,
+        });
+        this.downloadDocument(url);
+        // eslint-disable-next-line
+        await new Promise((resolve) => setTimeout(resolve, 2000));
+      }
+      notification.success({
+        message: `Successfully Downloaded: ${totalFiles} files.`,
+        duration: 10,
+      });
+
+      this.props.setNoticeOfWorkApplicationDocumentDownloadState({
+        downloading: false,
+        currentFile: 1,
+        totalFiles: 1,
+      });
+      this.props.closeModal();
+    });
+  };
+
+  openDownloadPackageModal = (event) => {
+    event.preventDefault();
+    const column = this.props.type === "REF" ? "is_referral_package" : "is_consultation_package";
+    const coreDocumentsInPackage = this.props.noticeOfWork.documents
+      .filter((document) => document[column])
+      .map(({ now_application_document_xref_guid }) => now_application_document_xref_guid);
+
+    const submissionDocumentsInPackage = this.props.noticeOfWork.filtered_submission_documents
+      .filter((document) => document[column])
+      .map(({ mine_document_guid }) => mine_document_guid);
+
+    this.props.openModal({
+      width: 910,
+      props: {
+        noticeOfWorkGuid: this.props.noticeOfWork.now_application_guid,
+        submissionDocuments: this.props.noticeOfWork.filtered_submission_documents,
+        importNowSubmissionDocumentsJob: this.props.importNowSubmissionDocumentsJob,
+        coreDocuments: this.props.noticeOfWork.documents,
+        onSubmit: this.downloadDocumentPackage,
+        cancelDownload: this.cancelDownload,
+        title: "Download Referral Package",
+        submissionDocumentsInPackage,
+        coreDocumentsInPackage,
+        handleSavePackage: this.handleSavePackage,
+        type: this.props.type,
+      },
+      content: modalConfig.DOWNLOAD_DOC_PACKAGE,
+    });
+  };
+
+  handleSavePackage = (selectedCoreRows, selectedSubmissionRows) => {
+    const column = this.props.type === "REF" ? "is_referral_package" : "is_consultation_package";
+    const documentsPayload = this.props.noticeOfWork.documents.map((document) => {
+      document[column] = selectedCoreRows.includes(document.now_application_document_xref_guid);
+      return document;
+    });
+
+    const submissionDocumentsPayload = this.props.noticeOfWork.filtered_submission_documents.map(
+      (document) => {
+        document[column] = selectedSubmissionRows.includes(document.mine_document_guid);
+        return document;
+      }
+    );
+
+    const payload = {
+      ...this.props.noticeOfWork,
+      documents: documentsPayload,
+      submission_documents: submissionDocumentsPayload,
+    };
+
+    const message =
+      this.props.type === "REF"
+        ? "Successfully updated the Referral Package."
+        : "Successfully updated the Consultation Package.";
+
+    return this.props
```

Comment:
Happy to see the returning of these actions :)

Reply:
:)